libcloud/test/compute/test_ec2.py

```diff
@@ -1548,6 +1553,26 @@ def _DescribeVolumes(self, method, url, body, headers):
         body = self.fixtures.load('describe_volumes.xml')
         return (httplib.OK, body, {}, httplib.responses[httplib.OK])
 
+    def _filters_DescribeVolumes(self, method, url, body, headers):
+        expected_params_1 = {
+            'Filter.1.Name': 'attachment.instance-id',
+            'Filter.1.Value.1': 'i-d334b4b3',
+        }
+
+        expected_params_2 = {
+            'Filter.1.Name': 'status',
+            'Filter.1.Value.1': 'available'
+        }
+
+        try:
+            self.assertUrlContainsQueryParams(url, expected_params_1)
```

Comment:
Eventually we should improve this utility method so the order doesn't matter :)

Reply:
I just checked and see the method already supports that.

It's just the way tests are structured. This should do it - f8deceb04c41e36bc3002b3fd9c4f0d6e752b383.